packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -686,19 +801,24 @@ describe('TransactionController', () => {
           },
         ];
 
+        // this feels like a weird thing to test here
         const pendingTransactions =
           nonceTrackerMock.mock.calls[0][0].getPendingTransactions(
             ACCOUNT_MOCK,
           );
 
-        expect(nonceTrackerMock).toHaveBeenCalledTimes(1);
+        // gets called in constructor now
+        expect(nonceTrackerMock).toHaveBeenCalledTimes(4);
         expect(pendingTransactions).toStrictEqual([
           expect.any(Object),
           ...externalPendingTransactions,
         ]);
         expect(getExternalPendingTransactions).toHaveBeenCalledTimes(1);
         expect(getExternalPendingTransactions).toHaveBeenCalledWith(
           ACCOUNT_MOCK,
+          // This is undefined for the base nonceTracker
+          // TODO (AD) add tests for using external pending transactions with a networkClientId once we have trackingMaps instantiated
```

Comment:
this is covered in integration tests

Reply:
https://github.com/MetaMask/core/pull/3643/commits/c1b6466a5a863fce996bf4195b9fdc77caa40713